src/utils.ts

```diff
@@ -110,6 +110,42 @@ export function getAxisLabels(arr: ZarrArray, axis_labels?: string[]): [...strin
   return axis_labels as [...string[], 'y', 'x'];
 }
 
+export function getNgffAxes(multiscales: Ome.Multiscale[]): Ome.Axis[] {
+  // Returns axes in the latest v0.4+ format.
+  // defaults for v0.1 & v0.2
+  const default_axes = [
+    { type: 'time', name: 't' },
+    { type: 'channel', name: 'c' },
+    { type: 'space', name: 'z' },
+    { type: 'space', name: 'y' },
+    { type: 'space', name: 'x' },
+  ];
+  function getDefaultType(name: string): string {
+    if (name === 't') return 'time';
+    if (name === 'c') return 'channel';
+    return 'space';
+  }
+  let axes = default_axes;
+  // v0.3 & v0.4+
+  if (multiscales[0].axes) {
+    axes = multiscales[0].axes.map((axis) => {
+      // axis may be string 'x' (v0.3) or object
+      let name: string = axis instanceof String ? (axis as string) : (axis as Ome.Axis).name;
+      let type = getDefaultType(name);
+      if (!(axis instanceof String) && (axis as Ome.Axis).type) {
+        type = (axis as Ome.Axis).type as string;
+      }
+      return { name, type };
+    });
+  }
+  return axes;
```

Comment:
Lots of type assertions here which are an escape-hatch from the TS type checker (so if the types ever change this might break). Perhaps something like this would be more explicit/safe:

```suggestion
  if (multiscales[0].axes) {
    axes = multiscales[0].axes.map((axis) => {
      // axis may be string 'x' (v0.3) or object
      if (typeof axis === 'string') {
        return { name: axis, type: getDefaultType(axis) }
      }
      const { name, type } = axis;
      return { name, type: type ?? getDefaultType(name) };
    });
  }
```